libs/filamat/src/GLSLPostProcessor.cpp

```diff
@@ -38,6 +40,41 @@ using namespace spvtools;
 
 namespace filamat {
 
+namespace {
+
+    std::string addMaybeUnusedToVariables(const std::string& mslShader) {
+        constexpr std::string_view unusedVariables[] = {
+            "float3x3 shading_tangentToWorld;",
+            "float3 shading_position;",
+            "float3 shading_view;",
+            "float3 shading_normal;",
+            "float3 shading_geometricNormal;",
+            "float3 shading_reflected;",
+            "float shading_NoV;",
+            "float2 shading_normalizedViewportCoord;",
+            "float3 shading_clearCoatNormal;",
+            "float filament_lodBias = frameUniforms.lodBias;",
+            "float3 bentNormal = param_8;"
+        };
+
+        constexpr std::string_view maybeUnused = "[[maybe_unused]] ";
+
+        std::string patchedMslShader;
+        patchedMslShader.reserve(std::size(maybeUnused) * std::size(unusedVariables) + std::size(mslShader));
+        patchedMslShader = mslShader;
+        for (const auto& variable : unusedVariables) {
+
+            auto it = std::search(std::begin(patchedMslShader), std::end(patchedMslShader), std::begin(variable), std::end(variable));
+            if (it != std::end(patchedMslShader)) {
+                patchedMslShader.insert(it, std::begin(maybeUnused), std::end(maybeUnused));
+            }
```

Comment:
nit: `string::find(string_view)` and `string::insert(size_t, string_view)` would allow for less bloat

Reply:
https://github.com/shapr3d/filament/pull/100/commits/79fc0935f24e75651cf23c5de23f7bd5e73ea1ae